src/main/java/com/commercetools/project/sync/BaseSyncStatisticsDeserializer.java

```diff
@@ -0,0 +1,39 @@
+package com.commercetools.project.sync;
+
+import com.commercetools.sync.commons.helpers.BaseSyncStatistics;
+import com.commercetools.sync.customers.helpers.CustomerSyncStatistics;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import java.io.IOException;
+
+public class BaseSyncStatisticsDeserializer extends StdDeserializer<BaseSyncStatistics> {
+
+  public BaseSyncStatisticsDeserializer() {
+    this(null);
+  }
+
+  public BaseSyncStatisticsDeserializer(final Class<?> vc) {
+    super(vc);
+  }
+
+  @Override
+  public BaseSyncStatistics deserialize(JsonParser jsonParser, DeserializationContext ctxt)
+      throws IOException, JsonProcessingException {
+    final ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
+    final JsonNode syncStatisticsNode = mapper.readTree(jsonParser);
+    // since BaseSyncStatistics is abstract and there's no difference in the Subclasses except
+    // implementation of
+    // method getReportMessage(), additionally there's no field which could be used to identify the
+    // type of the
+    // subclass, the jsonNode is mapped to any concrete class
+    // this is a workaround, to fix that BaseSyncStatistics in java-sync library needs to be
+    // adjusted declaring
+    // the subtypes with Json annotations: JsonTypeInfo and JsonSubTypes
+    // more details: https://www.baeldung.com/jackson-inheritance#2-per-class-annotations
+    return mapper.treeToValue(syncStatisticsNode, CustomerSyncStatistics.class);
```

Comment:
CustomerSyncStatistics.class ? what about for other Statistics classes? 
I don't think it is the right approach to change BaseSyncStatistics.(I mean adding JsonTypeInfo and JsonSubTypes to the library class, Since it is used by different users not only Project-Sync).

Did you try implementing deserializer for Generic type?  
https://stackoverflow.com/questions/36159677/how-to-create-a-custom-deserializer-in-jackson-for-a-generic-type/36223769
may be something like this? WDYT?

Reply:
As described this is a workaround as BaseSyncStatistics doesn't support JsonTypeInfo and JsonSubTypes atm. Why do you think changing the library class is a problem?

Haven't tried implementing deserializer for generic type. Might solve the problem. I'll try if this approach will work in this case.

Reply:
@praveenkumarct The problem here is, as `BaseSyncStatistics` is abstract and `<T extends BaseSyncStatistics>` in context or beanproperty the type is always `BaseSyncStatistics` but never actual subtypes. So, this approach doesn't solve the problem.

Reply:
Thank you Sarah for trying with the generic approach. May be you can create a task or issue to remove this workaround later.